gamestonk_terminal/portfolio/rh_api.py

```diff
@@ -0,0 +1,164 @@
+import argparse
+import datetime
+from datetime import datetime as dt
+import pandas as pd
+from pandas.plotting import register_matplotlib_converters
+import matplotlib.pyplot as plt
+import mplfinance as mpf
+import numpy as np
+
+import robin_stocks.robinhood as r
+from termcolor import colored
+from gamestonk_terminal.helper_funcs import (
+    parse_known_args_and_warn,
+    plot_autoscale,
+)
+from gamestonk_terminal import feature_flags as gtff
+from gamestonk_terminal.config_terminal import RH_USERNAME as user, RH_PASSWORD as pw
+
+register_matplotlib_converters()
+
+dt_format = "%Y-%m-%dT%H:%M:%SZ"
+valid_span = ["day", "week", "month", "3month", "year", "5year", "all"]
+valid_interval = ["5minute", "10minute", "hour", "day", "week"]
+
+span_title_dict = {
+    "day": "Day",
+    "week": "Week",
+    "month": "Month",
+    "3month": "3 Months",
+    "year": "Year",
+    "5year": "5 Years",
+    "all": "All Time",
+}
+
+
+def login():
+    r.login(user, pw)
+
+
+def logoff():
+    r.logout()
+
+
+def show_holdings():
+
+    holds = r.account.build_holdings()
+    stocks = []
+    equity = []
+    for stock, data in holds.items():
+        stocks.append(stock)
+        equity.append(round(float(data["equity"]), 2))
+
+    print("")
+    print("Stonk\t last price \t prev close \t equity \t % Change")
+    print("")
+    for stonk, eq in zip(stocks, equity):
+        stonk_data = r.stocks.get_quotes(stonk)[0]
+        prev_close = round(float(stonk_data["adjusted_previous_close"]), 2)
+        last_price = round(float(stonk_data["last_trade_price"]), 2)
+        pct_change = round((last_price - prev_close) / prev_close, 3)
+        to_print = f"{stonk}\t {last_price}\t\t {prev_close}\t\t {eq}\t\t {pct_change}"
+
+        if last_price >= prev_close:
+            print(colored(to_print, "green"))
+        else:
+            print(colored(to_print, "red"))
+
+    print("")
+
+
+def plot_historical(l_args):
```

Comment:
Same history about changing code to be more readable and modular. So, some things that would be advised in this area are:

- One function retrieves data, one function plots, and the other process the command
- Type hinting, e.g. `l_args` would be `other_args: List[str]`
- Docstring with numpy format